contentcuration/contentcuration/viewsets/sync/endpoint.py

```diff
@@ -210,11 +237,18 @@ def sync(request):
                 else:
                     viewset = viewset_class(request=request)
                     viewset.initial(request)
-                    es, cs = apply_changes(
-                        request, viewset, change_type, id_attr, changes
-                    )
-                    errors.extend(es)
-                    changes_to_return.extend(cs)
+                    if change_type in event_handlers:
+                        try:
+                            es, cs = event_handlers[change_type](
+                                request, viewset, id_attr, changes
+                            )
+                            errors.extend(es)
+                            changes_to_return.extend(cs)
+                        except Exception as e:
+                            errors.extend(changes)
+                            # Capture exception and report, but allow sync
+                            # to complete properly.
+                            report_exception(e)
```

Comment:
Why are we trying to avoid the 500 here? Especially when Sentry is not active (e.g. in dev environments), won't this result in 500s going unnoticed unless one checks the logs for them? Or does Sentry not report the 500 without this code?

Reply:
I'd be fine with throwing the 500 here in dev environments, but in production it might put people in a difficult state, can rethrow in the case we are in debug mode.

Reply:
Rethrowing in debug is fine for now, basically my immediate concern is that problems get caught as early as possible in testing.